testsuite/integration/elytron-oidc-client/src/test/java/org/wildfly/test/integration/elytron/oidc/client/propagation/OidcIdentityPropagationTestCase.java

```diff
@@ -0,0 +1,823 @@
+     * The EAR used in this test case contains:
+     * - a servlet within a WAR
+     * - a remote EJB within a JAR
+     *
+     * The servlet attempts to invoke the remote EJB.
+     */
+    @Test
+    @OperateOnDeployment(SINGLE_DEPLOYMENT_REMOTE)
+    public void testServletToEjbSingleDeploymentRemote() throws Exception {
+        testServletToEjbInvocation();
+    }
+
+    /**
+     * The EAR used in this test case contains:
+     * - a servlet within a WAR
+     * - an EJB within a JAR
+     *
+     * The wrong password is used to access the servlet.
+     */
+    @Test
+    @OperateOnDeployment(SINGLE_DEPLOYMENT_REMOTE)
+    public void testWrongPasswordSingleDeployment() throws Exception {
+        testWrongPassword();
+    }
+
+    private void testServletToEjbInvocation() throws Exception {
+        String expectedMessage = "alice,true"; // alice should have Managers role
+        loginToApp(KeycloakConfiguration.ALICE, KeycloakConfiguration.ALICE_PASSWORD, HttpURLConnection.HTTP_OK, expectedMessage,
+                true, url.toURI().resolve("whoAmI"));
+    }
+
+    private void testWrongPassword() throws Exception {
+        loginToApp(KeycloakConfiguration.ALICE, "WRONG_PASSWORD", HttpURLConnection.HTTP_OK, "Invalid username or password", true, url.toURI().resolve("whoAmI"));
+    }
+
+    /**
+     * The EAR used in this test case contains:
+     * - a servlet within a WAR
+     * - a local EJB within a JAR
+     *
+     * The servlet attempts to invoke the local EJB.
+     * The identity being used to invoke the servlet doesn't exist in the security
+     * domain that's being used to secure the EJB.
+     */
+    @Test
+    @OperateOnDeployment(SINGLE_DEPLOYMENT_LOCAL)
+    public void testServletToEjbSingleDeploymentOutflowNotPossibleNonExistentIdentity() throws Exception {
+        String expectedMessage = "anonymous,false"; // charlie won't be outflowed, the target security domain's current identity will be used instead
+        loginToApp(KeycloakConfiguration.CHARLIE, KeycloakConfiguration.CHARLIE_PASSWORD, HttpURLConnection.HTTP_OK, expectedMessage, true, url.toURI().resolve("whoAmI"));
+    }
+
+    /**
+     * The EAR used in this test case contains:
+     * - a servlet within a WAR
+     * - a local EJB within a JAR
+     *
+     * The servlet attempts to invoke the local EJB.
+     * The security domain that's being used to secure the EJB hasn't been configured to
+     * trust the virtual security domain.
+     */
+    @Test
+    @OperateOnDeployment(ANOTHER_SINGLE_DEPLOYMENT_LOCAL)
+    public void testServletToEjbSingleDeploymentOutflowNotPossibleTrustNotConfigured() throws Exception {
+        String expectedMessage = "anonymous,false"; // charlie won't be outflowed, the target security domain's current identity will be used instead
+        loginToApp(KeycloakConfiguration.CHARLIE, KeycloakConfiguration.CHARLIE_PASSWORD, HttpURLConnection.HTTP_OK, expectedMessage, true, url.toURI().resolve("anotherWhoAmI"));
```

Comment:
IIUIC, another user should be used (e.g. Alice) as Charlie _doesn't exist in the security domain that's being used to secure the EJB_.